cmd/image-builder/logging.go

```diff
@@ -59,27 +64,34 @@ func requestIdExtractMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 			iid = random.String(12)
 		}
 
+		// create fields stored with every log statement
+		rd := logrus.Fields{
+			"method": c.Request().Method,
+			"path":   c.Path(),
+		}
+		for _, key := range c.ParamNames() {
+			// protect existing and the most important fields
+			if _, ok := rd[key]; !(ok || key == "msg" || key == "level") {
+				rd[key] = c.Param(key)
+			}
+		}
+
 		// store it in a standard context
 		ctx := c.Request().Context()
 		ctx = context.WithValue(ctx, requestIdCtx, rid)
 		ctx = context.WithValue(ctx, insightsRequestIdCtx, iid)
+		ctx = context.WithValue(ctx, requestDataCtx, rd)
 		c.SetRequest(c.Request().WithContext(ctx))
 
-		f := logrus.Fields{"method": c.Request().Method, "path": c.Path()}
-		for _, key := range c.ParamNames() {
-			f[key] = c.Param(key)
-		}
-
 		// and set echo logger to be context logger
 		ctxLogger := logrus.StandardLogger()
 		c.SetLogger(&common.EchoLogrusLogger{
 			Logger: ctxLogger,
 			Ctx:    ctx,
-			Fields: f,
 		})
 
 		if !SkipPath(c.Path()) {
-			c.Logger().Debugf("Started request")
+			ctxLogger.WithContext(ctx).WithFields(rd).Debugf("Started request")
```

Comment:
This hunk is probably unnecessary, I was experimenting around to find the best solution.

Reply:
```suggestion
			ctxLogger.WithContext(ctx).Debugf("Started request")
```
my `WithFields` is also not needed here, right?

Reply:
It is either this, or I can make use of the echo logger created above. Let me use the echo logger instead, looks better.